sermant-backend/src/main/java/io/sermant/backend/service/ConfigService.java

```diff
@@ -69,66 +68,56 @@ public class ConfigService {
-        if (!result.isSuccess()) {
-            return new Result<>(result.getCode(), result.getMessage());
-        }
-        String requestGroup = request.getGroup();
+    public Result<List<ConfigInfo>> getConfigList(ConfigInfo request) {
         ConfigClient client = getConfigClient(request.getNamespace());
-        if (client instanceof NacosClient) {
-            requestGroup = NacosUtils.rebuildGroup(requestGroup);
+        if (client == null || !client.isConnect()) {
+            return new Result<>(ResultCodeType.CONNECT_FAIL);
+        }
+        Result<PluginTemplateInfo> templateInfoResult = templateService.getTemplate(request.getPluginType());
+        if (!templateInfoResult.isSuccess() || templateInfoResult.getData() == null) {
+            return new Result<>(ResultCodeType.FAIL.getCode(), "Invalid plugin name.");
         }
-        Map<String, List<String>> configMap = client.getConfigList(request.getKey(), requestGroup, exactMatchFlag);
+        PluginTemplateInfo template = templateInfoResult.getData();
+        List<String> keyRules = StringUtils.isEmpty(request.getKeyRule()) ?
+                template.getKeyRule() : Collections.singletonList(request.getKeyRule());
+        List<String> groupRules = StringUtils.isEmpty(request.getGroupRule()) ?
+                template.getGroupRule() : Collections.singletonList(request.getGroupRule());
         List<ConfigInfo> configInfoList = new ArrayList<>();
-        PluginConfigHandler handler = pluginType.getHandler();
-        for (Map.Entry<String, List<String>> entry : configMap.entrySet()) {
-            String group = entry.getKey();
-            if (client instanceof NacosClient) {
-                group = NacosUtils.convertGroup(group);
-            } else if (client instanceof KieClient) {
-                group = group.replace(KieConstants.CONNECTOR, KieConstants.SEPARATOR);
-                group = group.replace(KieConstants.DEFAULT_LABEL_PRE, StringUtils.EMPTY);
-            }
-            if (!exactMatchFlag && !handler.verifyConfigurationGroup(group)) {
-                continue;
-            }
-            for (String configKey : entry.getValue()) {
-                if (!exactMatchFlag && !handler.verifyConfigurationKey(configKey)) {
-                    continue;
+        groupRules.forEach(groupRule -> {
+            String groupPattern = replaceWithWildcard(groupRule);
+            Map<String, List<String>> configMap = client.getConfigList(CommonConst.WILDCARD, groupPattern, false);
+            configMap.forEach((group, value) -> value.forEach(key -> keyRules.forEach(keyRule -> {
+                String keyPattern = replaceWithWildcard(keyRule);
+                if (Pattern.compile(keyPattern).matcher(key).matches()) {
+                    ConfigInfo configInfo = new ConfigInfo(key, group, keyRule, groupRule, request.getPluginType(),
+                            request.getNamespace());
+                    configInfoList.add(configInfo);
                 }
```

Comment:
exacted to a method

Reply:
There are too many parameters. The extraction method is not recommended.